tests/test_check.py

```diff
@@ -212,3 +213,31 @@ def test_check_no_reporter_output(prevent_webbrowser_and_httpserver) -> None:
     # If the file exists, the assertion failed and the file gets removed from main directory
     if file_exists:
         remove("pyta_output.html")
+
+
+def test_check_all_output_typing_io() -> None:
+    """Test that specifying output as a typing.IO stream captures the output report when check_all is called."""
+    output = io.StringIO()
+
+    python_ta.check_all(
+        "examples/custom_checkers/e9989_pep8_errors.py",
+        config={"output-format": "python_ta.reporters.PlainReporter"},
+        output=output,
+    )
+
+    assert "E9989 (pep8-errors)" in output.getvalue()
+    output.close()
```

Comment:
While the overall approach here is good, if the above assertion fails then `output.close()` will not be called.

A better approach is to define a simple pytest fixture [using `yield`](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended) that will handle the creation of an `io.StringIO` and then clean it up at the end of the test regardless of whether the test passes or fails.